```
2019-01-29T02:16:37.3786826Z ________________________________________________________________________________
2019-01-29T02:16:37.3787011Z _____________ sympy/matrices/tests/test_commonmatrix.py:test_trace _____________
2019-01-29T02:16:37.3787160Z Traceback (most recent call last):
2019-01-29T02:16:37.3787674Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/matrices/tests/test_commonmatrix.py", line 547, in test_trace
2019-01-29T02:16:37.3787875Z     assert M.trace() == 14
2019-01-29T02:16:37.3788511Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/matrices/common.py", line 1829, in trace
2019-01-29T02:16:37.3789921Z     if not self.is_square:
2019-01-29T02:16:37.3790232Z AttributeError: 'OperationsOnlyMatrix' object has no attribute 'is_square'
2019-01-29T02:16:37.3790324Z ________________________________________________________________________________
2019-01-29T02:16:37.3790383Z _____________ sympy/matrices/tests/test_commonmatrix.py:test_power _____________
2019-01-29T02:16:37.3790428Z Traceback (most recent call last):
2019-01-29T02:16:37.3790772Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/matrices/tests/test_commonmatrix.py", line 704, in test_power
2019-01-29T02:16:37.3790831Z     assert (A**5)[:] == (6140, 8097, 10796, 14237)
2019-01-29T02:16:37.3791141Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/core/decorators.py", line 132, in binary_op_wrapper
2019-01-29T02:16:37.3791223Z     return func(self, other)
2019-01-29T02:16:37.3791525Z   File "/opt/hostedtoolcache/Python/2.7.15/x64/lib/python2.7/site-packages/sympy-1.4.dev0-py2.7.egg/sympy/matrices/common.py", line 2052, in __pow__
2019-01-29T02:16:37.3791579Z     if not self.is_square:
2019-01-29T02:16:37.3791977Z AttributeError: 'ArithmeticOnlyMatrix' object has no attribute 'is_square'
```

Some primitive matrix classes don't have property `is_square`, while `rows` and `cols` are defined in every matrices

--- a/sympy/matrices/common.py
+++ b/sympy/matrices/common.py
@@ -1826,7 +1826,7 @@ class MatrixOperations(MatrixRequired):
         5
 
         """
-        if not self.is_square:
+        if not self.rows == self.cols:
             raise NonSquareMatrixError()
         return self._eval_trace()
 
@@ -2049,7 +2049,7 @@ class MatrixArithmetic(MatrixRequired):
 
     @call_highest_priority('__rpow__')
     def __pow__(self, num):
-        if not self.is_square:
+        if not self.rows == self.cols:
             raise NonSquareMatrixError()
         try:
             a = self

--- a/sympy/matrices/common.py
+++ b/sympy/matrices/common.py
@@ -1826,7 +1826,7 @@ class MatrixOperations(MatrixRequired):
         5
 
         """
-        if not self.rows == self.cols:
+        if self.rows != self.cols:
             raise NonSquareMatrixError()
         return self._eval_trace()
 
@@ -2049,7 +2049,7 @@ class MatrixArithmetic(MatrixRequired):
 
     @call_highest_priority('__rpow__')
     def __pow__(self, num):
-        if not self.rows == self.cols:
+        if self.rows != self.cols:
             raise NonSquareMatrixError()
         try:
             a = self

